app/views/handlers/list_action.py

```diff
@@ -80,9 +83,7 @@ def handle_post(self):
         )
 
         if self.questionnaire_store_updater.is_dirty():
-            self._routing_path = self.router.routing_path(
-                self.current_location.section_id, self.current_location.list_item_id
-            )
+            self._routing_path = self._get_routing_path()
```

Comment:
When it was working out which block to route to next, it used list item id in the routing path, which then meant it checked every block in the section with (block_id, list_item_id) rather than (block_id, None) which would be the correct key for a non-repeating section, my thought was that the routing path ought to be based on the list collector rather than its children, but I'm happy to look into another way of fixing this issue if it shouldn't

Reply:
I'm not sure that I fully understand the scenario. Might be a good one to discuss/demo.

Reply:
As discussed have removed the list item id from the method this calls and left this as is